pyrsistent/_pmap.py

```diff
@@ -125,7 +125,27 @@ def __len__(self):
     def __repr__(self):
         return 'pmap({0})'.format(str(dict(self)))
 
-    __eq__ = Mapping.__eq__
+    def __eq__(self, other):
+        if self is other:
+            return True
+        if not isinstance(other, Mapping):
+            return NotImplemented
+        if len(self) != len(other):
+            return False
+        if isinstance(other, PMap):
+            try:
+                if hash(self) != hash(other):
+                    return False
+            except TypeError:
+                # self or other contains non-hashable values
+                pass
+            if self._buckets == other._buckets:
+                return True
+            return dict(self.iteritems()) == dict(other.iteritems())
```

Comment:
So in the worst case (equal map but with different number of buckets that does not have a hash cached) we would now traverse the structure three times instead of one as before to determine equality? Once to generate the hash, once over the buckets and once when iterating over the items?

Did you do any worst case measurements of this implementation against the old one?